src/Control/Monad/Logic/Sequence/Compat.hs

```diff
@@ -1,5 +1,27 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE ViewPatterns #-}
 module Control.Monad.Logic.Sequence.Compat
   ( fromSeqT
   , toLogicT
-  , fromLogicT ) where
-import Control.Monad.Logic.Sequence.Internal
+  , fromLogicT
+  , observeT
+  , observe ) where
+
+import Control.Monad.Identity (Identity(..))
+import Control.Monad.Logic.Sequence.Internal hiding ( observeT, observe )
+
+#if !MIN_VERSION_base(4,13,0)
+observeT :: Monad m => SeqT m a -> m a
+#else
+observeT :: MonadFail m => SeqT m a -> m a
+#endif
+observeT (toView -> m) = m >>= go where
+  go (a :< _) = return a
+  go Empty = fail "No results."
```

Comment:
Shouldn't the failure message identify the function?

Reply:
This is for `Compat` and this is the definition that is in `LogicT`.

/shrug